src/Symfony/Component/Clock/Resources/now.php

```diff
@@ -15,27 +15,14 @@
     /**
      * @throws \DateMalformedStringException When the modifier is invalid
      */
-    function now(string $modifier = null): \DateTimeImmutable
+    function now(string $modifier = 'now'): TimePoint
```

Comment:
Well, here we provide an _extra_ guarantee that our own clock always returns this TimePoint and not other implementations.

Reply:
Is the extra guarantee _necessary_?

Reply:
nothing is necessary, but this still moves the needle forward by allowing eg static analyzers to know that some situations can be handled in a simpler way (eg not checking for `false`, catching exceptions, etc.)

Reply:
That's very pragmatic and it makes perfect sense for a framework to do so.
The only question I have in that case: are we still honoring PSR20 in spirit?

Reply:
yes of course. Code written against the PSR20 ClockInterface will not benefit from the extra guarantee (because it works against the PSR20 interface that does not offer it). But we satisfy the PSR20 signature.

Reply:
So a Mock on a TimePoint would expect callers to be using Symfony\Clock or Symfony\MockClock. Makes perfect sense in my book.

That works well for the test.
However, in a project when the "timepoint for now" is being injected as a service or lazy proxy, should developers be instructed to adhere to Psr\Clock and \DateTimeImmutable or Symfony\...\Clock and ...\TimePoint?

That's the only part I'm worried about that seems to rattle the cage on encapsulation and abstraction creep.

Reply:
To keep it practical: should the language on the Clock docs be adjusted with this implementation detail?

https://github.com/symfony/symfony-docs/blob/6.4/components/clock.rst

Reply:
We should definitely tell about the new class in the documentation, but what is written is still accurate.